File: internal/content/testdata/bench/go/pairs-with-same-mod.go
func count_pairs_same_mod(nums []int, divisor int) int {
	freq := make(map[int]int)
	for _, x := range nums {
		freq[x%divisor]++
	}
	pairs := 0
	for _, cnt := range freq {
		pairs += cnt * (cnt - 1) / 2
	}
	return pairs
}

File: internal/content/testdata/bench/rust/pairs-with-same-mod.rs
use std::collections::HashMap;

fn count_pairs_same_mod(nums: Vec<i64>, divisor: i64) -> i64 {
    let mut freq: HashMap<i64, i64> = HashMap::new();
    for x in &nums {
        let key = x.rem_euclid(divisor);
        *freq.entry(key).or_insert(0) += 1;
    }
    let mut pairs: i64 = 0;
    for cnt in freq.values() {
        pairs += cnt * (cnt - 1) / 2;
    }
    pairs
}
